frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -798,22 +836,366 @@ class SegmentsView extends React.Component<Props, State> {
+    return this.state != null && this.doesGroupHaveAnyMeshes(groupId)
+      ? {
+          key: "removeMeshes",
+          label: (
+            <div
+              onClick={() => {
+                this.handleRemoveMeshes(groupId);
+                this.handleSegmentDropdownMenuVisibility(groupId, false);
+              }}
+            >
+              <DeleteOutlined />
+              Remove all meshes
+            </div>
+          ),
+        }
+      : null;
+  };
+
+  getDownLoadMeshesItem = (groupId: number): ItemType => {
+    return this.state != null && this.doesGroupHaveAnyMeshes(groupId)
+      ? {
+          key: "downloadAllMeshes",
+          label: (
+            <div
+              onClick={() => {
+                this.downloadAllMeshesForGroup(groupId);
+                this.handleSegmentDropdownMenuVisibility(groupId, false);
+              }}
+            >
+              <DownloadOutlined />
+              Download all meshes
+            </div>
+          ),
+        }
+      : null;
+  };
+
+  getMoveSegementsHereItem = (groupId: number): ItemType => {
+    return this.state.selectedSegmentId != null
+      ? {
+          key: "moveHere",
+          onClick: () => {
+            if (
+              this.state.selectedSegmentId == null ||
+              this.props.visibleSegmentationLayer == null
+            ) {
+              // Satisfy TS
+              return;
+            }
+            this.props.updateSegment(
+              this.state.selectedSegmentId,
+              { groupId },
+              this.props.visibleSegmentationLayer.name,
+              true,
+            );
+            this.handleSegmentDropdownMenuVisibility(groupId, false);
+          },
+          disabled: !this.props.allowUpdate,
+          icon: <ArrowRightOutlined />,
+          label: "Move active segment here",
+        }
+      : null;
+  };
+
+  getShowMeshesItem = (groupId: number): ItemType => {
+    return this.state != null && this.doesGroupHaveAnyMeshes(groupId)
+      ? {
+          key: "showMeshesOfGroup",
+          label: (
+            <div
+              onClick={() => {
+                if (this.props.visibleSegmentationLayer == null) {
+                  // Satisfy TS
+                  return;
+                }
+                this.handleChangeMeshVisibilityInGroup(
+                  this.props.visibleSegmentationLayer.name,
+                  groupId,
+                  !this.state.areSegmentsInGroupVisible[groupId],
+                );
+                this.handleSegmentDropdownMenuVisibility(groupId, false);
+              }}
+            >
+              <i className="fas fa-dice-d20" />
+              {this.state.areSegmentsInGroupVisible[groupId] ? "Hide" : "Show"} all meshes
+            </div>
+          ),
+        }
+      : null;
+  };
+
+  setGroupColor(groupId: number, color: Vector3) {
+    const { visibleSegmentationLayer } = this.props;
+    if (visibleSegmentationLayer == null) return;
```

Comment:
I thought it would be nice to avoid any further computation of the `groupToSegmentMap` if the `visibleSegmentationLayer` is null anyway. But you're saying this doesn't have much impact?

Reply:
Good point. I did not think about the compute cost. I guess it does not matter much either way. Fine to leave as is.